banyand/stream/tag.go

```diff
@@ -68,6 +73,22 @@ func (t *tag) mustWriteTo(tm *tagMetadata, tagWriter *writer) {
 	}
 	tm.offset = tagWriter.bytesWritten
 	tagWriter.MustWrite(bb.Buf)
+
+	if t.filter != nil {
+		bb.Reset()
+		var err error
+		bb.Buf, err = encoding.BloomFilterToBytes(t.filter)
```

Comment:
```suggestion
		bb.Buf, err = encoding.BloomFilterToBytes(bb.Buf[:0], t.filter)
```

You should pass bb.Buf into this function to avoid object allocation.